pkg/controllers/collaset/synccontrol/update.go

```diff
@@ -253,23 +265,26 @@ func (o orderByDefault) Less(i, j int) bool {
 }
 
 type PodUpdater interface {
+	PrepareAndFilterPodUpdate(podToUpdate []*PodUpdateInfo, resources *collasetutils.RelatedResources,
```

Comment:
Two points I find need to discuss:
1. I think `PrepareAndFilterPodUpdate` could be split into two methods: `FilterPodToUpdate` and `PreparePodUpdate`
2. `Recreate` and `InPlaceIfPossible` have same implementation. I think they can share the same top Updater and be dispatched by another Interface.